File: packages/app/src/services/software/definitions/ccleaner.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'ccleaner';
	public name = 'CCleaner';
	public category = [SoftwareCategory.FileManagement, SoftwareCategory.SystemManagement, SoftwareCategory.Utility];
	public downloadName = 'ccsetup.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = true;
	public deprecated = false;
	public icon = 'ccleaner.png';
	public homepage = 'https://ccleaner.com';

	public async resolveDownloadUrl() {
		return ok('https://bits.avcdn.net/productfamily_CCLEANER/insttype_FREE/platform_WIN_PIR/installertype_ONLINE/build_RELEASE');
	}
}

File: src/lib/software/ccleaner.py
from src.lib.software import BaseSoftware

class Ccleaner(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'ccsetup.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://bits.avcdn.net/productfamily_CCLEANER/insttype_FREE/platform_WIN_PIR/installertype_ONLINE/build_RELEASE')
